### PR TITLE
[docs] Runtime Core fixes

### PR DESCRIPTION
**Pull request by wateret**
- Fix some grammar mistakes
- Fix empty links
- Reflect feedback from #2607

ONE-DCO-1.0-Signed-off-by: Hanjoung Lee <hanjoung.lee@samsung.com>

Thanks to @ragmani

### CODE REVIEW

File: docs/runtime/core.md [hunk 1]
```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
...
+Here's a figure of how those data structures are organized.
 
 ![Core](core-figure-ir.png)
 
 ### Module `compiler`
 
 `Compiler` is the main class of this module. Everything starts from it.
 
-What it does is making models executable. It schedules run order and assigns a backend for each operation. Here are major phases of compilation.
+What it does is making models executable. It schedules execution order and assigns a backend for each operation. Here are major phases of compilation.
```
Comment (wateret) [inline docs/runtime/core.md:37]: @ragmani From #2607

File: docs/runtime/core.md [hunk 2]
```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
...
 
 #### 1. Lowering
 
-In Lowering, the compiler assigns a [backend](#) for each operation. A backend is assigned to an operation means that the operation will be run with the assigned backend's kernel.
+In Lowering, `Compiler` assigns a [backend](#) for each operation. It means that the operation will be run with the assigned backend's kernel.
```
Comment (wateret) [inline docs/runtime/core.md:41]: @ragmani From #2607 

And I replaced the whole subject with pronoun 'it' since it looks redundant 😄

File: docs/runtime/core.md [hunk 3]
```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
...
 
 `Operand` and `Operation` are nodes of the graph, and the reference relationship between them is the edges of the graph.
 
-`Subgraphs` represents the whole model. It could have more than one `Subgraph` to support control flow operations. Those operations can make calls to another subgraph and when the execution on another subgraph is done it gets back to previous subgraph execution with returned operands.
+`Subgraphs` represents the whole model. A model can have more than one `Subgraph` to support control flow operations. Those operations make calls to another subgraph and when the execution on another subgraph is done it gets back to previous subgraph execution with returned operands.
 
 All graphs are a [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) so once model inputs are given we can run it in topological order.
 
-Here's a figure of how those data structures are oraganized.
+Here's a figure of how those data structures are organized.
```
Comment (wateret) [inline docs/runtime/core.md:29]: @ragmani From #2607

File: docs/runtime/core.md [hunk 4]
```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
 Runtime Core has four modules. These are namespaces as well as directory names in `/runtime/onert/core/src/`.
 
 - `ir`  stands for Intermediate Representation which contains Neural Network Graph data structures
-- `compiler` converts IR to executable format
-- `exec` is execution module which is the result of compilation
-- `backend` is backend interface
+- `compiler` converts IR to an executable format
+- `exec` is an execution module which is the result of a compilation
+- `backend` is a backend interface
```
Comment (hyunsik-yoon) [inline docs/runtime/core.md:12]: seems like a recursive definition. :-)
I guess  we need some explanation of _backend_. (People who have experience with server may think `database` when they hear `backend`. :-P)

 I found this in this doc. How about using it here?
```
Backend interface classes are mostly about memory management and kernel generation.
```
or something like this?

_backend is provides API and implementation related to real calculation of operation. Some of its roles are providing execution logic optimzied for a specific hardware such as CPU, GPU or NPU, and memory management._

Comment (wateret) [reply docs/runtime/core.md:12]: This is a very short summary for so I would like to keep it as a one-liner. I may add a bit more words for that later. (I won't make change for that in this PR as it is not relevant.)

File: docs/runtime/core.md [hunk 5]
```diff
@@ -7,72 +7,72 @@ Runtime Core is a compilation/execution engine for neural network models.
...
+'kernel' here means an implementation of the actual calculation of an operation.
 
 A backend is assigned for each operation. In this phase, a kernel for each operation is generated.
 
 Let's say we have some functions written in a certain programming language. Then its compiler compiles each function into a chunk of assembly. Here 'function' is like 'operation' and 'assembly' is like 'kernel'.
 
 #### 5. Create Executor
 
-With generated tensors and kernels, the compiler creates executor objects. There are 3 types of executors are supported - Linear, Dataflow and Parallel. Linear executor is the default executor and Dataflow Executor and Parallel Executor are experimental.
+With generated tensors and kernels, the compiler creates executor objects. There are 3 types of executors are supported - Linear, Dataflow, and Parallel. Linear executor is the default executor and Dataflow Executor and Parallel Executor are experimental.
```
Comment (ragmani) [inline docs/runtime/core.md:69]: Why did you add `,` after `Dataflow`?

Comment (wateret) [reply docs/runtime/core.md:69]: Hmm.. I also thought it was strange but Grammarly made this change 🙄
